### PR TITLE
Added events for adding and filtering profile summaries

### PR DESCRIPTION
**Pull request by tsteur**
@sgiehl adding events similar to visitor details events. Will be needed in next release if possible.

### CODE REVIEW

File: plugins/Live/Controller.php [hunk 1]
```diff
@@ -228,11 +228,42 @@ public static function getAllProfileSummaryInstances()
...
+             * **Example**
+             *
+             *     public function addProfileSummary(&$profileSummaries)
+             *     {
+             *         $profileSummaries[] = new MyCustomProfileSummary();
+             *     }
+             *
+             * @param ProfileSummaryAbstract[] $profileSummaries An array of profile summaries
+             */
+            Piwik::postEvent('Live.addProfileSummaries', array(&$instances));
```
Comment (sgiehl) [inline plugins/Live/Controller.php]: Not sure if we should introduce an event for adding summaries, as this is done automatically by looking for classes extending the `ProfileSummary` class.

Comment (tsteur) [reply plugins/Live/Controller.php]: It is basically the same as for Visitor Details, Widgets, Reports, ... this way, it allows plugins that define entities, such as custom dimensions, to add for example a profile summary for each entity / custom dimension. It is just a general thing that we do so far as not always it is possible to have it in the plugin structure.

**Review by sgiehl — APPROVED** [review]

File: plugins/Live/Controller.php (hunk 1 above)
Comment (sgiehl) [reply plugins/Live/Controller.php]: ah sure. Guess I haven't thought enough before writing the comment...

**Comment by tsteur** [comment]
UI test failures seem to be random failures by the looks or not related to this PR:  eg https://builds-artifacts.piwik.org/piwik/piwik/3.x-dev/25168/UIIntegrationTest_dashboard2.png

**Comment by sgiehl** [comment]
Those failures are related to #12220 I guess

**Pull request merged**